RequestKit/Router.swift

```diff
@@ -43,7 +43,12 @@ public protocol Router {
 public extension Router {
     public func request() -> NSURLRequest? {
         let URLString = configuration.apiEndpoint.stringByAppendingURLPath(path)
-        var parameters = encoding == .JSON ? [:] : params
+        var parameters: [String: String] = [:]
+        if encoding != .JSON {
```

Comment:
Thanks, you’re absolutely right about the fact that the `params` property’s type shouldn’t be relaxed. Instead, in f76a2e7580b52e542da38c0dd4ce053cb2758fb5, I’ve added a specialized `jsonParams` property directly to `JSONPostRouter`. `JSONPostRouter` prefers `jsonParams` but will fall back to `params` if `jsonParams` is `nil`.

I also removed the tests but left in the modifications to `TestRouter`, as a way of ensuring that URL- and JSON-encoded cases can coexist in the same route.